src/app/shared/services/course.service.ts

```diff
@@ -1,128 +1,119 @@
+        params.set('count', limit.toString());
+        if (!!search) {
+            params.set('query', search);
+        }
+        requestOptions.search = params;
+
+        this.sub = this.http.get(this.serviceUrl, requestOptions)
+            .map(this.mapData)
+            .map(this.mapToCourses)
+            .catch((error) => {
+                console.error('error', error);
+                return Observable.throw(error);
+            })
+            .subscribe((courses: ICourse[]) => {
+                console.log('loaded courses:', courses);
+                this.courseListSubject.next(courses);
+            });
     }
 
     create(newCourse: INewCourse): Observable<ICourse> {
         let timestamp: number = new Date().getTime();
         let course: ICourse = Object.assign(newCourse, {
-            id: `uuid${timestamp}`,
-            topRated: false,
+            id: timestamp,
+            isTopRated: false,
             date: new Date(newCourse.date)
         });
 
-        this.courseList.push(course);
-        this.courseListSubject.next(this.courseList.slice());
+        console.warn('This method doesn\'t work now');
+
+        // this.courseList.push(course);
+        // this.courseListSubject.next(this.courseList.slice());
 
         return Observable.of<ICourse>(course).delay(DELAY);
     }
 
-    getById(id: string): Observable<ICourse> {
-        return Observable.of<ICourse>(this.courseList.find((item: ICourse) => item.id === id)).delay(DELAY);
+    getById(id: number): Observable<ICourse> {
+        return this.http.get(`${this.serviceUrl}/${id}`)
+            .map(this.mapData)
+            .map(this.mapToCourse)
+            .catch((error) => {
+                console.error('error', error);
+                return Observable.throw(error);
+            });
     }
 
-    update(id: string): Observable<ICourse> {
-        let course: ICourse = this.courseList.find((item: ICourse) => item.id === id);
-        let updateCourse: any = {name: `${course.name} [edited]`};
+    update(id: number): Observable<ICourse> {
+        console.warn('this method doesn\'t work now');
 
-        if (course) {
-            const updated = Object.assign({}, course, updateCourse);
-            this.courseList.splice(this.courseList.findIndex((item) => item.id === id), 1, updated);
-
-            this.courseListSubject.next(this.courseList.slice());
-            return Observable.of<ICourse>(updated).delay(DELAY);
-        }
         return Observable.of<ICourse>(null);
     }
 
-    remove(id: string): Observable<boolean> {
-        this.courseList.splice(this.courseList.findIndex((item) => item.id === id), 1);
-        this.courseListSubject.next(this.courseList.slice());
-        return Observable.of(true).delay(DELAY);
+    remove(id: number): Observable<boolean> {
+        return this.http.delete(`${this.serviceUrl}/${id}`)
+            .map(this.mapData)
+            .map((res) => {
+                return true;
```

Comment:
Niby tak, generalnie `.map(this.mapData)` używam do mapowania na jsona przy wszystkich requestach. A drugie raczej powinienem sprawdzać status z serwera, ale wolałem to tak zostawić, żeby jak coś na przyszłość dodać tutaj dodatkową logikę